File: Upcoder/Lập trình cơ bản/3.Vòng lặp/Tam giác sao vuông cân/TAMGIACSAO2.cs
using System;

namespace TAMGIACSAO
{
    public class Program
    {
        public static void Main(string[] args)
        {
            long n = long.Parse(Console.ReadLine());
            long k = n;

            for (long i = 1; i <= n; i++)
            {
                k -= i;
                if (k <= 0)
                    break;
            }

            if (k == 0)
                Console.Write("YES");
            else
                Console.Write("NO");
        }
    }
}

File: Upcoder/Lập trình cơ bản/3.Vòng lặp/Tam giác sao vuông cân/TAMGIACSAO2.cpp
#include<iostream>
using namespace std;
int main()
{
    long long n;
    cin >> n;
    long long k = n;
    for(long long i = 1; i <= n; i++)
    {
        k = k - i;
        if(k <= 0)
            break;
    }
    if(k == 0)
        cout << "YES";
    else
        cout << "NO";
}